app/services/CompositeUserService.scala

```diff
@@ -0,0 +1,26 @@
+package services
+
+import model.User
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CompositeUserService(okta: UserService, identityDb: LegacyUserService)(implicit ctx: ExecutionContext)
+    extends UserService {
+
+  def healthCheck(): Future[Unit] = for {
+    _ <- okta.healthCheck()
+    _ <- identityDb.healthCheck()
+  } yield ()
+
+  def fetchUserByIdentityId(id: String): Future[Option[User]] = for {
+    optOktaUser <- okta.fetchUserByIdentityId(id)
+    optLegacyUser <- identityDb.fetchByIdentityId(id)
+  } yield {
+    (optOktaUser, optLegacyUser) match {
+      case (Some(oktaUser), Some(legacyUser)) =>
+        Some(oktaUser.copy(userName = legacyUser.userName, permissions = legacyUser.permissions))
+      case (Some(oktaUser), None) => Some(oktaUser)
```

Comment:
should we return a user if it doesn't exist in the legacy repository ? I think in other places we consider this a broken user and ignore it.

Reply:
OK I'll follow the same convention then.